File: app/src/main/java/cz/lsrom/webviewtest/QrScannerActivity.java
package cz.lsrom.webviewtest;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;

import com.google.zxing.Result;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import butterknife.BindView;
import butterknife.ButterKnife;
import me.dm7.barcodescanner.zxing.ZXingScannerView;

/**
 * @author Lukas Srom <lukas.srom@gmail.com>
 */
public class QrScannerActivity extends AppCompatActivity implements ZXingScannerView.ResultHandler {
    private static final String TAG = QrScannerActivity.class.getSimpleName();

    @BindView(R.id.qr_scanner_view)
    protected ZXingScannerView scannerView;

    public static Intent getStartIntent (@NonNull Context context){
        return new Intent(context, QrScannerActivity.class);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_qr_scanner);
        ButterKnife.bind(this, this);
    }

    @Override
    public void onPause() {
        super.onPause();
        if (scannerView != null) {
            scannerView.stopCamera();
        }
    }

    @Override
    public void onResume() {
        super.onResume();
        if (scannerView != null) {
            scannerView.setResultHandler(this); // Register ourselves as a handler for scan results.
            scannerView.startCamera();          // Start camera on resume
        }
    }

    @Override
    public void handleResult(Result rawResult) {
        String result = rawResult.getText();
        Log.d(TAG, result);

        startActivity(MainActivity.getStartIntent(result, this));
    }


}

File: app/src/main/kotlin/cz/lsrom/webviewtest/qr/ui/QrScannerActivity.kt
/*
 * Copyright © 2019, Oracle and/or its affiliates. All rights reserved.
 */

package cz.lsrom.webviewtest.qr.ui

import android.Manifest
import android.content.Intent
import android.content.pm.PackageManager
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat
import androidx.core.view.isVisible
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.google.zxing.Result
import cz.lsrom.webviewtest.R
import cz.lsrom.webviewtest.app.MainActivity
import cz.lsrom.webviewtest.webview.platform.WebViewPresenter
import kotlinx.android.synthetic.main.qr_scanner_view.*
import me.dm7.barcodescanner.zxing.ZXingScannerView
import timber.log.Timber
import java.net.URL

const val URL_EXTRA = "url"

class QrScannerActivity : AppCompatActivity(),
    ZXingScannerView.ResultHandler {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.qr_scanner_view)

        if (!permissionGranted()) {
            requestPermission()
        }

        request_permission.setOnClickListener {
            requestPermission()
        }
    }

    private fun permissionGranted(): Boolean {
        return ContextCompat.checkSelfPermission(
            this,
            Manifest.permission.CAMERA
        ) == PackageManager.PERMISSION_GRANTED
    }

    private fun requestPermission() {
        ActivityCompat.requestPermissions(
            this,
            arrayOf(Manifest.permission.CAMERA),
            666
        )
    }

    public override fun onPause() {
        super.onPause()
        if (permissionGranted()) {
            qr_scanner_view.stopCamera()
        }
    }

    public override fun onResume() {
        super.onResume()
        if (permissionGranted()) {
            qr_scanner_view.setResultHandler(this)
            qr_scanner_view.startCamera()
        }
    }

    override fun handleResult(rawResult: Result) {
        startActivity(activityIntent(rawResult.text))
    }

    private fun activityIntent(url: String): Intent{
        return Intent(baseContext, MainActivity::class.java).apply {
            putExtra(URL_EXTRA, url)
        }
    }

    override fun onRequestPermissionsResult(
        requestCode: Int,
        permissions: Array<out String>,
        grantResults: IntArray
    ) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        Timber.d("request permission result")
        if (grantResults[0] == PackageManager.PERMISSION_DENIED) {
            request_permission.isVisible = true
            qr_scanner_view.isVisible = false
        } else {
            request_permission.isVisible = false
            qr_scanner_view.isVisible = true
            qr_scanner_view.startCamera()
        }
    }
}
